pkg/cache/scheduler/fair_sharing.go

```diff
@@ -38,17 +39,81 @@ type dominantResourceShareNode interface {
 	hierarchicalResourceNode
 }
 
-// dominantResourceShare returns a value ranging from 0 to math.MaxInt,
+// DRS contains the DominantResourceShare for some
+// node, with convenence methods for precise comparison.
+type DRS struct {
+	fairWeight       *resource.Quantity
+	unweightedRatio  float64
+	dominantResource corev1.ResourceName
+}
+
+// NegativeDRS is used as a starting point for comparisons.
+func NegativeDRS() DRS {
+	return DRS{unweightedRatio: -1, dominantResource: "", fairWeight: &oneQuantity}
+}
+
+// IsZero returns whether the DRS is 0. In other words,
+// the node for which this function was called is not
+// borrowing any resources.
+func (d DRS) IsZero() bool {
+	return d.unweightedRatio == 0
+}
+
+func (d DRS) PreciseWeightedShare() float64 {
+	if d.IsZero() {
+		return 0.0
+	}
+	if d.fairWeight.IsZero() {
+		return math.Inf(1)
+	}
+	return float64(d.unweightedRatio) * 1000.0 / float64(d.fairWeight.MilliValue())
+}
+
+// CompareDRS compares two DRS values. A lower value
+// indicates that the ClusterQueue/Cohort with this
+// value should be preferred for scheduling, while
+// a higher value preferred for preemption.
+func CompareDRS(a, b DRS) int {
```

Comment:
Any particular reason why this is not implementing the comparable interface, like `func (a DRS) Compare(b DRS) int {`?

Reply:
ack